pkg/resource/bucket/hook.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package bucket
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/s3"
+)
+
+func (rm *resourceManager) syncPutFields(
+	ctx context.Context,
+	r *resource,
+) (synced *resource, err error) {
+	// Copy resource for later patching
+	synced = &resource{r.ko.DeepCopy()}
+
+	diff, err := rm.diffLogging(ctx, synced)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(diff.Differences) > 0 {
+		rm.syncLogging(ctx, synced)
+	}
+
+	return synced, nil
+}
+
+func (rm *resourceManager) diffLogging(
+	ctx context.Context,
+	r *resource,
+) (*ackcompare.Delta, error) {
+	desired := r.ko.Spec.Logging
+
+	input := &svcsdk.GetBucketLoggingInput{
+		Bucket: r.ko.Spec.Name,
+	}
+	latest, err := rm.sdkapi.GetBucketLoggingWithContext(ctx, input)
```

Comment:
`latest` and `desired` here will be of different types, correct?

It would not cause trouble to compare their primitive fields but shouldn't they both be `ko.Spec.Logging`? Currently `latest` is type from aws-sdk-go and not ACK.

Reply:
Agreed. @RedbackThomson we typically use the variable name `resp` for the response object from the SDK call. Recommend using `resp` here.